apis/kueue/v1beta1/clusterqueue_types.go

```diff
@@ -98,10 +98,16 @@ type ClusterQueueSpec struct {
 	// +kubebuilder:default={}
 	Preemption *ClusterQueuePreemption `json:"preemption,omitempty"`
 
-	// admissionChecks lists the AdmissionChecks required by this ClusterQueue
+	// admissionChecks lists the AdmissionChecks required by this ClusterQueue.
+	// Cannot be used along with AdmissionCheckStrategy.
 	// +optional
 	AdmissionChecks []string `json:"admissionChecks,omitempty"`
 
+	// admissionCheckStrategy defines a list of strategies to determine which ResourceFlavors require AdmissionChecks.
+	// This property cannot be used in conjunction with the 'admissionChecks' property.
+	// +optional
+	AdmissionChecksStrategy AdmissionChecksStrategy `json:"admissionChecksStrategy,omitempty"`
```

Comment:
```suggestion
	AdmissionChecksStrategy *AdmissionChecksStrategy `json:"admissionChecksStrategy,omitempty"`
```

Based on [the API documentation](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required), I'm wondering if we should make this field pointer.

@alculquicondor WDYT?

Reply:
oh right, now that it's not a list, it's worth making it a pointer.